test/e2e_node/gpu_device_plugin.go

```diff
@@ -146,19 +150,28 @@ func restartKubelet(f *framework.Framework) {
 	kube := matches[0]
 	framework.Logf("Get running kubelet with systemctl: %v, %v", string(stdout), kube)
 	stdout, err = exec.Command("sudo", "systemctl", "restart", kube).CombinedOutput()
-	if err == nil {
-		return
+	if err != nil {
+		framework.Failf("Failed to restart kubelet with systemctl: %v, %v", err, stdout)
 	}
-	framework.Failf("Failed to restart kubelet with systemctl: %v, %v", err, stdout)
+	Eventually(func() bool {
+		afterSocks, err := filepath.Glob("/var/lib/kubelet/device-plugins/nvidiaGPU*.sock")
+		if err != nil || len(afterSocks) < 1 {
+			return false
+		}
+		framework.Logf("beforeSock %s afterSock %s numberOfGpus %d", beforeSocks[0], afterSocks[0], framework.NumberOfNVIDIAGPUs(getLocalNode(f)))
+		return (afterSocks[0] != beforeSocks[0] && framework.NumberOfNVIDIAGPUs(getLocalNode(f)) > 0)
+	}, 5*time.Minute, time.Second).Should(BeTrue())
 }
 
 func getDeviceId(f *framework.Framework, podName string, contName string, restartCount int32) string {
 	// Wait till pod has been restarted at least restartCount times.
 	Eventually(func() bool {
 		p, err := f.PodClient().Get(podName, metav1.GetOptions{})
-		framework.ExpectNoError(err)
+		if err != nil || len(p.Status.ContainerStatuses) < 1 {
+			return false
+		}
 		return p.Status.ContainerStatuses[0].RestartCount >= restartCount
-	}, time.Minute, time.Second).Should(BeTrue())
+	}, 5*time.Minute, time.Second).Should(BeTrue())
```

Comment:
nit: use `framework.Poll` for the poll interval

Reply:
I didn't see a framework.Poll function. Do you mean wait.Poll()? I can change to use wait.Poll() + ExpectNoError but it seems Eventually() has been used widely in our testing code.

Reply:
It's not a function, it's a constant used as the polling interval
https://github.com/kubernetes/kubernetes/blob/f0a061e361bccd7080e6adb0674b0c02582608e8/test/e2e/framework/util.go#L122
E.g.
https://github.com/kubernetes/kubernetes/blob/f0a061e361bccd7080e6adb0674b0c02582608e8/test/e2e/common/configmap.go#L295

Reply:
Ah sorry I misunderstood your previous comment. Thanks a lot for the explanation!